lib/msal-common/src/cache/CacheHelpers.ts

```diff
@@ -56,114 +56,121 @@ export class CacheHelpers {
     }
 
     /**
      * Reset all temporary cache items
      * @param state
      */
-    resetTempCacheItems(state?: string): void {
+    async resetTempCacheItems(state?: string): Promise<void> {
         // check state and remove associated cache items
-        this.cacheStorage.getKeys().forEach(key => {
+        const cacheKeys = await this.cacheStorage.getKeys();
+        const removedTempCacheItems: Array<Promise<void>> = cacheKeys.map(key => {
             if (!StringUtils.isEmpty(state) && key.indexOf(state) !== -1) {
                 const splitKey = key.split(Constants.RESOURCE_DELIM);
                 const keyState = splitKey.length > 1 ? splitKey[splitKey.length-1]: null;
                 if (keyState === state) {
-                    this.cacheStorage.removeItem(key);
+                    return this.cacheStorage.removeItem(key);
                 }
             }
-        });
+            return null;
+        }).filter(removed => removed);
+        await Promise.all(removedTempCacheItems);
         // delete generic interactive request parameters
-        this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_STATE);
-        this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_PARAMS);
-        this.cacheStorage.removeItem(TemporaryCacheKeys.ORIGIN_URI);
+        await this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_STATE);
+        await this.cacheStorage.removeItem(TemporaryCacheKeys.REQUEST_PARAMS);
+        await this.cacheStorage.removeItem(TemporaryCacheKeys.ORIGIN_URI);
     }
 
     /**
      * Get all access tokens in the cache
      * @param clientId
      * @param homeAccountIdentifier
      */
-    getAllAccessTokens(clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): Array<AccessTokenCacheItem> {
-        const results = this.cacheStorage.getKeys().reduce<Array<AccessTokenCacheItem>>((tokens, key) => {
-            const keyMatches = key.match(clientId) && key.match(authority) && key.match(resource) && key.match(homeAccountIdentifier);
-            if (keyMatches) {
-                const value = this.cacheStorage.getItem(key);
-                if (value) {
-                    try {
-                        const parseAtKey = JSON.parse(key) as AccessTokenKey;
-                        if (this.checkForExactKeyMatch(parseAtKey, clientId, authority, resource, homeAccountIdentifier)) {
-                            const newAccessTokenCacheItem = new AccessTokenCacheItem(parseAtKey, JSON.parse(value) as AccessTokenValue);
-                            return tokens.concat([ newAccessTokenCacheItem ]);
-                        }
-                    } catch (e) {
-                        throw ClientAuthError.createCacheParseError(key);
+    async getAllAccessTokens(clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): Promise<Array<AccessTokenCacheItem>> {
+        const cacheKeys: Array<string> = await this.cacheStorage.getKeys();
```

Comment:
Please review this code in depth, I had to make some changes and want to make sure it checks out. 
Tests passed. 

In interesting change is the change from reduce to map for promises